fe/fe-core/src/main/java/com/starrocks/server/MetadataMgr.java

```diff
@@ -688,11 +690,17 @@ public Statistics getTableStatistics(OptimizerContext session,
         // FIXME: In testing env, `_statistics_.external_column_statistics` is not created, ignore query columns stats from it.
         Statistics statistics = FeConstants.runningUnitTest ? null :
                 getTableStatisticsFromInternalStatistics(table, columns);
-        if (statistics == null || statistics.getColumnStatistics().values().stream().allMatch(ColumnStatistic::hasNonStats)) {
+        if (statistics == null ||
+                statistics.getColumnStatistics().values().stream().allMatch(ColumnStatistic::hasNonStats)) {
             session.setObtainedFromInternalStatistics(false);
-            Optional<ConnectorMetadata> connectorMetadata = getOptionalMetadata(catalogName);
-            return connectorMetadata.map(metadata -> metadata.getTableStatistics(
-                    session, table, columns, partitionKeys, predicate, limit)).orElse(null);
+            // Avoid `analyze table` to collect table statistics from metadata.
+            if (StatisticUtils.statisticTableBlackListCheck(table.getId())) {
+                return statistics;
```

Comment:
Is it appropriate to return statistics directly?  Do we need to rebuild the statistics based on columns

Reply:
`columngetTableStatisticsFromInternalStatistics` will set `default unknown statistics` for all all columns which not found.